### DISCUSSION TITLE
Mosh Support

### DISCUSSION THREAD

**Discussion by jarreds** [community/community#8856] [Codespaces]
I've been able to connect to Codespaces to use emacs/vim via SSH, and it works pretty well.

I'm interested in using [Mosh](https://mosh.org) to connect to Codespaces to eliminate some of the interactive latency in the terminal.

Has anyone explored this setup? If so, what have you found?

**Comment by asciimike** [comment]
Given `Mosh will log the user in via SSH, then start a connection on a UDP port between 60000 and 61000` ([docs](https://mosh.org/#usage)), I assume that we aren't able to handle this since I don't think we support forwarding UDP ports out of codespaces (non-HTTP ports are on the list, but it's a little ways in the future).

I would take a look at https://github.com/cli/cli/pull/4915 as that work will enable easier SSH access using your favorite tools, and may pave a path forward for Mosh support in the future (you can try to support it this way as well, I'm just not sure it'll make it beyond the SSH login step).
